.github/workflows/Build.yml

```diff
@@ -60,117 +60,86 @@ jobs:
-        run: composer install --prefer-dist --no-interaction
+      - name: Setup PHP Action
+        uses: shivammathur/setup-php@2.15.0
+        with:
+          php-version: ${{ matrix.php-versions }}
+          extensions: ${{ env.extensions }}
+          coverage: xdebug
+          tools: pecl, composer
 
+      # Install dependencies and handle caching in one go.
+      # Dependencies need to be installed to make sure the PHPUnit classes are recognized.
+      # @link https://github.com/marketplace/actions/install-composer-dependencies
+      - name: Install Composer dependencies
+        uses: "ramsey/composer-install@v2"
+        with:
+          composer-options: "${{ matrix.mode }}"
+
+      # begin of DEV
       - name: Check Syntax
-        if: ${{ matrix.mode == 'dev' }}
-        continue-on-error: ${{ matrix.php-versions != '8.0' }}
+        if: ${{ matrix.mode == '' }}
+        # continue-on-error: ${{ matrix.php-versions != '8.0' }}
         run: PHP_CS_FIXER_IGNORE_ENV=1 vendor/bin/php-cs-fixer fix --config=.php-cs-fixer.php --verbose --diff --dry-run
 
       - name: copy Env
-        if: ${{ matrix.mode == 'dev' }}
+        if: ${{ matrix.mode == '' }}
         run: |
           cp .github/workflows/.env.${{ matrix.sql-versions }} .env
 
       - name: Generate secure key & Optimize application & Migrate
-        if: ${{ matrix.mode == 'dev' }}
+        if: ${{ matrix.mode == '' }}
         run: |
           php artisan key:generate
           php artisan optimize
           php artisan migrate
 
       - name: Apply tests
-        if: ${{ matrix.mode == 'dev' }}
+        if: ${{ matrix.mode == '' }}
         run: XDEBUG_MODE=coverage vendor/bin/phpunit --verbose
 
       - name: Codecov
-        if: ${{ matrix.mode == 'dev' }}
+        if: ${{ matrix.mode == '' }}
         uses: codecov/codecov-action@v1
 
       - name: Make sure we can go backward
-        if: ${{ matrix.mode == 'dev' }}
+        if: ${{ matrix.mode == '' }}
         run: php artisan migrate:rollback
 
       # end of DEV
       # begin of DIST
-      - name: Install Composer dependencies (dist)
-        if: ${{ matrix.mode == 'dist' }}
-        run: composer install --prefer-dist --no-interaction --no-dev
 
       - name: Build Dist
-        if: ${{ matrix.mode == 'dist' }}
+        if: ${{ matrix.mode == '--no-dev' }}
         run: |
           make dist-clean
 
       - name: copy Env
-        if: ${{ matrix.mode == 'dist' }}
+        if: ${{ matrix.mode == '--no-dev' }}
         run: |
           cp .github/workflows/.env.${{ matrix.sql-versions }} Lychee/.env
 
       - name: setup MySQL Database with v3
-        if: ${{ matrix.mode == 'dist' }}
+        if: ${{ matrix.mode == '--no-dev' }}
         run: |
           mysql -uroot -proot homestead_test < .github/workflows/v3.sql
 
       - name: Generate secure key & Migrate & go backward
-        if: ${{ matrix.mode == 'dist' }}
+        if: ${{ matrix.mode == '--no-dev' }}
```

Comment:
If we removed the separation of dev and dist and integrated them a little we could use the artisan steps above for this to remove some more duplication. That's not a deal breaker though.